### PR TITLE
Golang extensions (WIP)

### PR DESCRIPTION
**Pull request by borys-cl**

### CODE REVIEW

File: extension/golang/schemas.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	}
+	return result
+}
+
+func (thisSchemasBinder *schemasBinder) Find(id string) goext.Schema {
+	manager := schema.GetManager()
+	for _, rawSchema := range manager.OrderedSchemas() {
+		if rawSchema.ID == id {
+			return bindSchema(thisSchemasBinder.rawEnvironment, rawSchema)
+		}
```
Comment (Zimnx) [inline extension/golang/schemas.go]: You can use: `schema, ok := manager.Schema(id)` instead manually iterating over sorted list

File: extension/golang/schemas.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+		return err
+	}
+
+	err = tx.Create(resourceData)
+
+	if err != nil {
+		return err
+	}
+
+	return tx.RawTransaction().Commit()
```
Comment (Zimnx) [inline extension/golang/schemas.go]: use `tx.Commit()` beacuse tx can be wrapped with some additional logic.

File: extension/golang/schemas.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+	return nil
+}
+
+//Create - creates resource
+func (thisSchemaBinder *schemaBinder) Create(resource interface{}) error {
+	tx, err := thisSchemaBinder.rawEnvironment.DataStore.Begin()
+
+	if err != nil {
+		return err
```
Comment (Zimnx) [inline extension/golang/schemas.go]: Each return branch causes tx to leak because it's not rolled back

File: extension/golang/schemas.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+	return tx.RawTransaction().Commit()
+}
+
+//Update - updates resource
+func (thisSchemaBinder *schemaBinder) Update(resource interface{}) error {
+	tx, err := thisSchemaBinder.rawEnvironment.DataStore.Begin()
+
+	if err != nil {
+		return err
```
Comment (Zimnx) [inline extension/golang/schemas.go]: tx leak

File: extension/golang/schemas.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+	}
+
+	err = tx.Update(resourceData)
+
+	if err != nil {
+		return err
+	}
+
+	return tx.RawTransaction().Commit()
```
Comment (Zimnx) [inline extension/golang/schemas.go]: use tx.Commit()

File: extension/golang/schemas.go [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+		return err
+	}
+
+	err = tx.Delete(thisSchemaBinder.rawSchema, id)
+
+	if err != nil {
+		return err
+	}
+
+	return tx.RawTransaction().Commit()
```
Comment (Zimnx) [inline extension/golang/schemas.go]: use `tx.Commit()`

File: extension/golang/schemas.go [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+	tx, err := thisSchemaBinder.rawEnvironment.DataStore.Begin()
+
+	if err != nil {
+		return err
+	}
+
+	err = tx.Delete(thisSchemaBinder.rawSchema, id)
+
+	if err != nil {
+		return err
```
Comment (Zimnx) [inline extension/golang/schemas.go]: tx leak

**Comment by borys-cl** [comment]
@Zimnx all fixed

**Pull request closed without merging**